source/manual/errors.html.md

```diff
@@ -8,9 +8,50 @@ last_reviewed_on: 2020-05-18
+      - Serve [a basic inline error page](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L531) if [all the mirrors fail](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L363) (uh oh!)
+
+    - For [other requests](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L367), we:
+      - [Serve the response as-is from the origin](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L374) (see below)
+
+  - When the origin servers are offline, we:
+      - [Skip normal request handling and begin error handling](https://varnish-cache.org/docs/trunk/users-guide/vcl-built-in-subs.html#vcl-backend-error)
+
+      - Try to [serve the page from a sequence of static mirrors](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L274)
+      - Serve [a basic inline error page](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L531) if [all the mirrors fail](https://github.com/alphagov/govuk-cdn-config/blob/d77313abdb5098e2b350de7a0992375e50ff03a3/vcl_templates/www.vcl.erb#L363) (uh oh!)
+
+  - When the origin servers return a 4xx response, we:
+      - Serve the response as-is from the origin (no special handling)
+
+2. **Origin**. When a request fails in an upstream app, we:
+
+  - [Intercept the response in Nginx](https://github.com/alphagov/govuk-puppet/blob/7dafec7cccd8308ec90c28835de70243d79b323b/modules/router/templates/router_include.conf.erb#L81)
+  - Replace the response with [a pre-rendered one from Static](https://github.com/alphagov/govuk-puppet/blob/7dafec7cccd8308ec90c28835de70243d79b323b/modules/router/manifests/errorpage.pp#L14)
+
+3. **App**. When an app raises an exception:
+
+  - If we have chosen to handle it, we:
+    - [Catch it and return a more helpful error response page](https://github.com/alphagov/email-alert-frontend/blob/a2bd35b5b17b7da40cd43df9c2756b564597b66e/app/controllers/application_controller.rb#L10)
+
+  - Otherwise, we:
+    - Let Rails return a "500 Internal Server Error" (by default)
+
+> Due to the way errors are handled by the origin servers, it's not currently possible for a frontend app to render a contextual error page for an error response. One option to get around this is to deviate from normal web semantics, and [return a 200 response](https://github.com/alphagov/email-alert-frontend/blob/a2bd35b5b17b7da40cd43df9c2756b564597b66e/app/controllers/email_alert_signups_controller.rb#L10). This should be avoided because such responses are cacheable, indexable (by search engines), and surprising for anyone trying to look for errors in logs.
```

Comment:
Is there a reason why these two paragraphs are indented with `>`? I understand it causes them to 'pop out' and draw attention to them - in which case, perhaps this should only apply to the second paragraph?

Reply:
I've made a few layout tweaks to clarify. They are both notes:

- The first is a note about app handling (last bullet point). I've replaced this with a possibly more convention "*" caveat.

- The second is a note about the whole section. I've moved this to the top, as it seems a bit pointless to find out after reading the section.